server/datastore/mysql/hosts.go

```diff
@@ -1916,6 +1922,17 @@ WHERE
 		osVersions.OSVersions = filtered
 	}
 
+	// filter by os versions by os id
+	if osID != nil {
```

Comment:
if `osID != nil` then `ds.OSVersions` will return always zero or one entry, right?

Reply:
Yes, the os id in the aggregated stats is derived from the id that is the primary key of the `operating_systems` table